roles/manage_transit_gateway/tasks/delete.yaml

```diff
@@ -0,0 +1,87 @@
+---
+- name: Run delete_tmanage_transit_gateway_gw_attach_vpc role
+  block:
+    - name: List all the transit gateway attachments
+      community.aws.ec2_transit_gateway_info:
+      register: manage_transit_gateway_info
+
+    - name: Get the transit gateway with the given description
+      ansible.builtin.set_fact:
+        manage_transit_gateway_gw: "{{ item }}"
+      when:
+        - item.description == transit_gateway.description
+        - item.state == "available"
+      loop: "{{ manage_transit_gateway_info.transit_gateways }}"
+
+    - name: Delete the attachment and transit manage_transit_gateway_gw
+      when: manage_transit_gateway_gw is defined
+      block:
+        - name: Describe attachments on a specific VPC
+          community.aws.ec2_transit_gateway_vpc_attachment_info:
+            filters:
+              transit-gateway-id: '{{ manage_transit_gateway_gw.transit_gateway_id }}'
+          register: manage_transit_gateway_info
+
+        - name: Start deletion of all attachments
+          community.aws.ec2_transit_gateway_vpc_attachment:
+            state: absent
+            id: '{{ item.transit_gateway_attachment_id }}'
+            wait: False
+          loop: '{{ manage_transit_gateway_info.attachments }}'
+
+        - name: Wait for the deletion of all attachments
+          community.aws.ec2_transit_gateway_vpc_attachment:
+            state: absent
+            id: '{{ item.transit_gateway_attachment_id }}'
+            wait: True
+          loop: '{{ manage_transit_gateway_info.attachments }}'
```

Comment:
I guess this one is probably redundant. You can probably wait in the previous task instead.